.azure-pipelines/ci-build-production.yml

```diff
@@ -16,7 +16,9 @@ pool:
 
 steps:
 
-- script: git checkout main
+- script: |
```

Comment:
any reason why we are not using the default `checkout` step?

Reply:
The checkout step failed 

https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=104684&view=logs&j=6b902995-b73d-5f5c-66fd-a7f66c857d2c&t=7cbfe919-8ac5-5a5d-dd4b-f8ecdf2baae9

Added the fetch step based on 
https://stackoverflow.com/a/7334027/10760931

Reply:
I meant like this:

https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/blob/89258fcb74ae1f58d632d3d4632c2efeab3d1d0f/.azure-pipelines/generation-templates/checkout-metadata.yml#L1

Reply:
@zengin  Can you please check and let me know if that works?